lib/hydra/works/indexers/collection_indexer.rb

```diff
@@ -0,0 +1,9 @@
+module Hydra::Works
+  class CollectionIndexer < Hydra::PCDM::CollectionIndexer
```

Comment:
That makes sense -- though I'm not sure we need to start the composition lower in the stack.  Could you pull out the generate_solr_document body, and let CollectionIndexer and GenericWork indexer call it?